roachtest: psycopg failed

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20009770?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20009770?buildTab=artifacts#/psycopg) on master @ [4344c0abdf85aa3836094dd2e7031d7f98175d55](https://github.com/cockroachdb/cockroach/commits/4344c0abdf85aa3836094dd2e7031d7f98175d55):


```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v25.3.0-alpha.4-dev-4344c0abdf85aa3836094dd2e7031d7f98175d55
Tests run against psycopg 3.2.8
3927 Total Tests Run
3926 tests passed
1 test failed
1919 tests skipped
31 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: tests.pool.test_pool_async.test_resize[asyncio] - unknown (unexpected)
For a full summary look at the psycopg artifacts 
An updated blocklist (psycopgBlockList) is available in the artifacts' psycopg log
test artifacts and logs in: /artifacts/psycopg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-20009770/psycopg/1750747018613/1750748012830)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51775

The failure we saw was:

```
_____________________________ test_resize[asyncio] _____________________________

dsn = 'host=localhost port=26257 user=roachprod password=cockroachdb dbname=defaultdb'

    @pytest.mark.slow
    @pytest.mark.timing
    async def test_resize(dsn):
        async def sampler():
            await asleep(0.05)  # ensure sampling happens after shrink check
            while True:
                await asleep(0.2)
                if p.closed:
                    break
                size.append(len(p._pool))
    
        async def client(t):
            async with p.connection() as conn:
                await conn.execute("select pg_sleep(%s)", [t])
    
        size: list[int] = []
    
        async with pool.AsyncConnectionPool(dsn, min_size=2, max_idle=0.2) as p:
            s = spawn(sampler)
    
            await asleep(0.3)
    
            c = spawn(client, args=(0.4,))
    
            await asleep(0.2)
            await p.resize(4)
            assert p.min_size == 4
            assert p.max_size == 4
    
            await asleep(0.4)
            await p.resize(2)
            assert p.min_size == 2
            assert p.max_size == 2
    
            await asleep(0.6)
    
        await gather(s, c)
>       assert size == [2, 1, 3, 4, 3, 2, 2]
E       assert [1, 1, 4, 4, 3, 2] == [2, 1, 3, 4, 3, 2, 2]
E         
E         At index 0 diff: 1 != 2
E         Right contains one more item: 2
E         
E         Full diff:
E           [
E         -     2,
E               1,
E         -     3,
E         ?     ^
E         +     1,
E         ?     ^
E         +     4,
E               4,
E               3,
E               2,
E         -     2,
E           ]

tests/pool/test_pool_async.py:683: AssertionError
```

--- a/pkg/cmd/roachtest/tests/psycopg.go
+++ b/pkg/cmd/roachtest/tests/psycopg.go
@@ -155,7 +155,7 @@ func registerPsycopg(r registry.Registry) {
 			cd /mnt/data1/psycopg/ &&
 			export PSYCOPG_TEST_DSN="host=localhost port={pgport:1} user=%[1]s password=%[2]s dbname=defaultdb" &&
 			export PGPASSWORD=%[2]s
-			pytest -vv --junit-xml=%[3]s`,
+			pytest -vv -m "not timing" --junit-xml=%[3]s`,
 			install.DefaultUser, install.DefaultPassword, testResultsXML))
 
 		// Fatal for a roachprod or transient error. A roachprod error is when result.Err==nil.

